webui/src/lib/components/controls.jsx

```diff
@@ -25,18 +25,18 @@ import {
 
 const defaultDebounceMs = 300;
 
-export function debounce(func, wait, immediate) {
+export const debounce = (func, wait, immediate) => {
     let timeout;
     return function() {
-        let context = this, args = arguments;
+        let args = arguments;
         let later = function() {
             timeout = null;
-            if (!immediate) func.apply(context, args);
+            if (!immediate) func.apply(null, args);
```

Comment:
This changes the semantics by a _lot_, now AFAIR `func` cannot be a method call!

Reply:
Not a JavaScript expert, but this is at odds with [this (random) StackOverflow answer](https://stackoverflow.com/a/33640254/192263).  Also MDN matches that answer and does not specify special behaviour for when `thisArg` is `null`.

To my understanding the new version is subtly different from the old one.  It's just that both will be subtly surprising in different ways when called from an object.  I'd rather not change it and just go for `_.debounce` (for some value of `_`), but I won't block on this.  Instead, we should just go over the actual current uses of `debounce` and see that none care about `this`.